docs/developer-guide/plugins.md

```diff
@@ -14,36 +14,47 @@ We recommend your custom rules adhere to our [rule conventions](rules.md) for:
 
 ## The anatomy of a plugin
 
+Let's say we write a plugin forbidding certain word (e.g., "foo") within a selector.
+The plugin's implementation would be:
+
 ```js
-// Abbreviated example
-const stylelint = require("stylelint");
+import stylelint from "stylelint";
+
+const { report, ruleMessages, validateOptions } = stylelint.utils;
 
-const ruleName = "plugin/foo-bar";
-const messages = stylelint.utils.ruleMessages(ruleName, {
-  expected: "Expected ..."
+const ruleName = "foo-org/selector-no-foo";
+
+const messages = ruleMessages(ruleName, {
+  rejected: (selector) => `Unexpected "foo" within selector "${selector}"`
 });
+
 const meta = {
-  url: "https://github.com/foo-org/stylelint-foo/blob/main/src/rules/foo-bar/README.md"
-  // deprecated: true,
+  url: "https://github.com/foo-org/stylelint-selector-no-foo/blob/main/README.md"
 };
 
-const ruleFunction = (primaryOption, secondaryOptionObject) => {
-  return (postcssRoot, postcssResult) => {
-    const validOptions = stylelint.utils.validateOptions(
-      postcssResult,
-      ruleName,
-      {
-        /* .. */
-      }
-    );
+const ruleFunction = (primary, secondaryOptions) => {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: primary,
+      possible: [true]
+    });
 
     if (!validOptions) {
       return;
     }
 
-    // ... some logic ...
-    stylelint.utils.report({
-      /* .. */
+    root.walkRules((ruleNode) => {
+      const { selector } = ruleNode;
+
+      if (selector.includes("foo")) {
+        report({
+          result,
+          ruleName,
+          message: messages.rejected(selector),
+          node: ruleNode
+          word: selector,
+        });
+      }
```

Comment:
```suggestion
      if (!selector.includes("foo")) return;
      
      report({
        result,
        ruleName,
        message: messages.rejected(selector),
        node: ruleNode
        word: selector,
      });
```

minor to advocate for our pattern of returning early in rules.

Reply:
Fixed bde97188d.